src/file_buffer.rs

```diff
@@ -135,7 +135,7 @@ impl<F: Write + Read + Seek> BufFile<F> {
     }
 
     fn fetch_slab(&mut self, mut loc: u64) -> Result<&mut Slab, Error> {
-        loc = loc & !SLAB_MASK;
+        loc = loc & SLAB_MASK;
```

Comment:
This seems wrong:
```rust
const SLAB_MASK: u64 = 0xFFFFF; // 1024 * 1024 - 1
let loc = 0x12345678;
println!("{}", loc & !SLAB_MASK); // 0x12300000
println!("{}", loc & SLAB_MASK); // 0x00045678
```

I think you really want the slab that starts at 0x12300000